fourth.py

```diff
@@ -6,6 +6,20 @@ def fibo_fun(n):
 
 print(fibo_fun(8))
 
+# to reduece time complexity
+
+cache_store = {0:1,1:1}
+
+def fibo_fun_witch_cache(n):
+    if n in {0,1}:
+        return cache_store[n]
+    else:
+        cache_store[n] = cache_store[n-1]+cache_store[n-2]
+        return cache_store[n]
+
+fibo_nums = [fibo_fun_witch_cache(n) for n in range(8)]
```

Comment:
This is iterative approach, not the recursive one. Please make the fibo_fun_withch_cache function recursive. If you don't know about recursion, please learn it again.